### ISSUE TITLE
Give `query` and `raw_query` of option for response syntax

### ISSUE THREAD

**Issue by Garllon** [Garllon/graphql_connector#20]
Give the 'raw_query' and the 'query' method the option to return the result as raw data hash or with openstructs. I think sometimes it is not necessary to create openstruct objects.

**Comment by sushie1984** [comment]
Do you plan to have smth like the following -->

```ruby
GraphqlConnector::<SERVER>.query('departments', {id: 1}, ['id', 'name'], as: Hash)

GraphqlConnector::<SERVER>.query('departments', {id: 1}, ['id', 'name'], as: OpenStruct)

GraphqlConnector::<SERVER>.raw_query('query { ... }', variables: { id: 1 }, as: Hash)

GraphqlConnector::<SERVER>.raw_query('query { ... }', variables: { id: 1 }, as: OpenStruct)
```

**Comment by Garllon** [comment]
Yeah something like this was my idea. The only problem is that we currently need to set a default for query its with Openstruct and for raw query the opposite. However this can then be changed with version 2.0. Currently i would go with a solution like yours here.

**Comment by sushie1984** [comment]
@Garllon 

**tldr**: Use `returns:` argument in much flexible way to avoid argument overflow on `query`, `add_query`, `raw_query`, `add_raw_query`, ...

**A bit longer**:
Another idea that came in my mind in order to use/extend arguments `returns:` of `query`/`add_query` for more flexible input (also adding a `returns` to  `raw_query`/`add_raw_query`:

```ruby
# Built-in default formatters --> Thus no breaking change when using 
# current [:id, :name, ...], but also as an option as an `returns` argument 
GraphqlConnector::HashFormat
GraphqlConnector::OpenStructFormat
[...]

GraphqlConnector::<SERVER>.query('departments', 
                                 {id: 1}, 
                                 GraphqlConnector::HashFormat.with(:id, :name, productCategory: [:id])

GraphqlConnector::<SERVER>.raw_query('query { ... }', 
                                     variables: { id: 1 }, 
                                     GraphqlConnector::OpenStructFormat.with(:id, :name, productCategory: [:id])

class DepartmentsQuery
   extend GraphqlConnector::<SERVER>::Query

   add_query by_id: :departments, 
             params: [:id], 
             returns: GraphqlConnector::HashFormat.with(:id, :name, productCategory: [:id]

   add_raw_ query by_id: 'query { ... }', 
                  params: [:id], 
                  returns: GraphqlConnector::OpenStructFormat.with(:id, :name, productCategory: [:id]
end
```

Or when preferring a custom format
```ruby

class MyCustomFormatter
  def self.with(**args)
     ...
  end
end

GraphqlConnector::<SERVER>.query('departments', 
                                 {id: 1}, 
                                 MyCustomFormatter.with(:id, :name, productCategory: [:id])
```

If implemented like this there might be a potential to use `returns` with [offset based pagination iteration](https://github.com/Garllon/graphql_connector/issues/22) as well, instead of putting another argument on those methods
